cypress/integration/class-selection-popover-online.ts

```diff
@@ -0,0 +1,401 @@
+  const mutationResult = await client.mutate({
+    mutation: gql`
+      mutation createImage($url: String, $datasetId: ID!) {
+        createImage(data: { url: $url, datasetId: $datasetId }) {
+          id
+          name
+          width
+          height
+          url
+        }
+      }
+    `,
+    variables: {
+      datasetId,
+      url,
+    },
+  });
+
+  const {
+    data: { createImage: image },
+  } = mutationResult;
+
+  return image;
+}
+
+const createLabel = (data: LabelCreateInput) => {
+  return client.mutate({
+    mutation: gql`
+      mutation createLabel($data: LabelCreateInput!) {
+        createLabel(data: $data) {
+          id
+        }
+      }
+    `,
+    variables: {
+      data,
+    },
+  });
+};
+
+const createLabelClass = async (
+  name: String,
+  color = "#ffffff",
+  datasetId: string
+) => {
+  const {
+    data: {
+      createLabelClass: { id },
+    },
+  } = await client.mutate({
+    mutation: gql`
+      mutation createLabelClass(
+        $name: String!
+        $color: String!
+        $datasetId: ID!
+      ) {
+        createLabelClass(
+          data: { name: $name, color: $color, datasetId: $datasetId }
+        ) {
+          id
+          name
+          color
+        }
+      }
+    `,
+    variables: {
+      name,
+      color,
+      datasetId,
+    },
+  });
+
+  return id;
+};
+
+describe("Class selection popover", () => {
+  let datasetId: string;
+  let datasetSlug: string;
+  let imageId: string;
+  beforeEach(() => {
+    cy.setCookie("consentedCookies", "true");
+    cy.task("performLogin").then((token) => {
+      cy.setCookie("next-auth.session-token", token as string);
+    });
+    cy.task("createWorkspaceAndDatasets").then(async () => {
+      const createResult = await createDataset("cypress test dataset");
+      datasetId = createResult.id;
+      datasetSlug = createResult.slug;
+
+      const { id } = await createImage(
+        "https://images.unsplash.com/photo-1579513141590-c597876aefbc?auto=format&fit=crop&w=882&q=80",
+        datasetId
+      );
+      imageId = id;
+
+      const labelClassId = await createLabelClass(
+        "A new class",
+        "#F87171",
+        datasetId
+      );
+      await createLabel({
+        imageId,
+        labelClassId,
+        type: LabelType.Box,
+        geometry: {
+          type: LabelType.Polygon,
+          coordinates: [
+            [
+              [0, 900],
+              [900, 900],
+              [900, 1500],
+              [0, 1500],
+              [0, 900],
+            ],
+          ],
+        },
+      });
+    });
+  });
+
+  it("right clicks on a label to change its class", () => {
+    // See https://docs.cypress.io/guides/core-concepts/conditional-testing#Welcome-wizard
+    cy.visit(
+      `/cypress-test-workspace/datasets/${datasetSlug}/images/${imageId}?modal-welcome=closed&modal-update-service-worker=update`
+    );
+    cy.get('[aria-label="loading indicator"]').should("not.exist");
```

Comment:
Wouldn't relying on `id` instead of `aria-label` be a safer way to get component instances? (🤷‍♂️ )

Reply:
The philosophy for why we use `aria-label` is that unless needed we shouldn't add code that is just useful for testing. `aria-label` are useful for accessibility plus they allow us to identify components so that's what we have used so far. I agree that it could become unsafe if we have repeated `aria-labels`, but for now it's not a problem since they are unique in a same view.